tests/scalers/loki/loki_test.go

```diff
@@ -0,0 +1,253 @@
+//go:build e2e
+// +build e2e
+
+package loki_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "loki-test"
+)
+
+var (
+	testNamespace         = fmt.Sprintf("%s-ns", testName)
+	deploymentName        = fmt.Sprintf("%s-deployment", testName)
+	monitoredAppName      = fmt.Sprintf("%s-monitored-app", testName)
+	publishDeploymentName = fmt.Sprintf("%s-publish", testName)
+	scaledObjectName      = fmt.Sprintf("%s-so", testName)
+	lokiServerName        = fmt.Sprintf("%s-server", testName)
+	minReplicaCount       = 0
+	maxReplicaCount       = 2
+)
+
+type templateData struct {
+	TestNamespace         string
+	DeploymentName        string
+	MonitoredAppName      string
+	PublishDeploymentName string
+	ScaledObjectName      string
+	LokiServerName        string
+	BackTick              string
+	MinReplicaCount       int
+	MaxReplicaCount       int
+}
+
+const (
+	deploymentTemplate = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: test-app
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: test-app
+  template:
+    metadata:
+      labels:
+        app: test-app
+        type: keda-testing
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+---
+`
+
+	monitoredAppDeploymentTemplate = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: {{.MonitoredAppName}}
+  name: {{.MonitoredAppName}}
+  namespace: {{.TestNamespace}}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{.MonitoredAppName}}
+  template:
+    metadata:
+      labels:
+        app: {{.MonitoredAppName}}
+        type: {{.MonitoredAppName}}
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+---
+`
```

Comment:
do we need this for something?

Reply:
If not, please also remove the related variables

Reply:
I saw it in the Prometheus scaler and thought it is necessary, however, yes, it is of any relevance here I think. I have removed monitoredAppDeployment altogether now.